client/src/main/java/io/quarkus/vault/client/auth/VaultToken.java

```diff
@@ -31,8 +32,15 @@ public VaultToken(String clientToken, boolean renewable, Duration leaseDuration,
         this.fromCache = fromCache;
     }
 
+    public VaultToken(String clientToken, boolean renewable, Duration leaseDuration, Instant created, boolean fromCache,
+            InstantSource instantSource) {
+        super(renewable, leaseDuration, created, instantSource);
+        this.clientToken = clientToken;
+        this.fromCache = fromCache;
+    }
+
     public VaultToken cached() {
-        return new VaultToken(clientToken, renewable, leaseDuration, true, instantSource);
+        return new VaultToken(clientToken, renewable, leaseDuration, created, true, instantSource);
```

Comment:
Please add a test case for (at least) this method:
```java
@Test
void cachedKeepsCreatedTimestamp() {
    VaultToken t1 =
        VaultToken.from("foo", true, Duration.ofMinutes(1), tickableInstanceSource);
    tickableInstanceSource.tick(Duration.ofSeconds(10));
    VaultToken t2 = t1.cached();
    assertEquals(t1.getExpiresAt(), t2.getExpiresAt());
}
```
(this would fail in v4.0.0)

Reply:
I added a higher-level test that ensures cached tokens are the same as those returned from the original token provider (except the cached flag).